server/src/main/java/org/apache/druid/server/QueryResource.java

```diff
@@ -565,4 +496,142 @@ public static void transferEntityTag(ResponseContext context, Response.ResponseB
       builder.header(HEADER_ETAG, entityTag);
     }
   }
+
+  private class QueryResourceQueryMetricCounter implements QueryMetricCounter
+  {
+    @Override
+    public void incrementSuccess()
+    {
+      successfulQueryCount.incrementAndGet();
```

Comment:
Nit: The different `AtomicLong`s for the counts can now be made member fields of this class itself.

Reply:
When you say "this class" which class are you talking about?  They are part of `QueryResource` because they are read by something somewhere...

Reply:
I mean move the fields to `QueryResourceQueryMetricCounter` as they are only ever incremented through the methods here and can also have `getters` here. 
But it was just a stye nitpick, it's perfectly fine either way.

Reply:
🤷  I see this as aligning with an interface/sharing state and so am not as bothered by not going through getters.  If this was in another class or something like that, I'd fully agree.